source-cn/behavioral/chain_of_responsibility.swift

```diff
@@ -57,24 +57,14 @@ final class MoneyPile: Withdrawing {
 
 final class ATM: Withdrawing {
 
-    private var hundred: Withdrawing
-    private var fifty: Withdrawing
-    private var twenty: Withdrawing
-    private var ten: Withdrawing
+    private var moneyPiles: [Withdrawing]
 
     private var startPile: Withdrawing {
-        return self.hundred
+        return self.moneyPiles.first!
```

Comment:
I appreciate your contribution. Force unwrap is not allowed, we are trying to teach folks good architecture, small infractions like that may give wrong impression that such code is acceptable, it's not.